pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -131,10 +131,20 @@ type ClusterWorkspaceTypeList struct {
 
 // ClusterWorkspaceInitializer is a unique string corresponding to a cluster workspace
 // initialization controller for the given type of workspaces.
-//
-// +kubebuilder:validation:MaxLength=32
-// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$`
-type ClusterWorkspaceInitializer string
+type ClusterWorkspaceInitializer struct {
+	// name is a unique string corresponding to a cluster workspace
+	// initialization controller for the given type of workspaces.
+	//
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$`
+	Name string `json:"name"`
+
+	// path is an absolute reference to the workspace that owns this initializer, e.g. root:org:ws.
+	// The workspace must be some ancestor or a child of some ancestor.
```

Comment:
copy and pasta.

As written above, this path in the type is unnecessary (and undesired). The type initializer path is just the lcluster of the type.

In the ClusterWorkspace we have to store the path though (and replicate into the labels somehow).

Reply:
We need to sync on this - I don't think I understand what you mean by "lcluster" if it's not the path ... ?

Reply:
lcluster = logical cluster = since recently also called workspace path.

Reply:
The "ancestor or a child ..." is the piece that looks like copy&paste.